### PR TITLE
Fix pagination issue

### PR DESCRIPTION
**Pull request by bdunne**
Fixes: PnT JIRA (TPINVTRY-461)

### CODE REVIEW

File: lib/topological_inventory/orchestrator/worker.rb [hunk 1]
```diff
@@ -132,7 +132,9 @@ def each_resource(url, tenant_account = ORCHESTRATOR_TENANT, &block)
         resources = paging ? response["data"] : response
         resources.each { |i| yield i }
 
-        each_resource(response.fetch_path("links", "next"), tenant_account, &block) if paging
+        return unless paging && next_page_link = response.fetch_path("links", "next")
```
Comment (carbonin) [inline lib/topological_inventory/orchestrator/worker.rb]: I think `next_page_path` would make the issue more explicit in the code here. The fact that we're only getting a path rather than an actual link is the problem we're trying to solve.

Also, multi condition `unless` one-liners are hard to parse.

Maybe this would be easier to read if we moved the assignment to the line above?

```ruby
next_page_path = response.fetch_path("links", "next")
return unless paging && next_page_path

next_url = URI.parse(url).merge(next_page_path).to_s
each_resource(next_url, tenant_account, &block)
```

and even then I still prefer the `if` version:
```ruby
return if !paging || next_page_path.nil?
```

Comment (bdunne) [reply lib/topological_inventory/orchestrator/worker.rb]: > I think next_page_path would make the issue more explicit in the code here.

I didn't want to use `path` in the variable name, because it's not just a path, it's a relative URL.  In our case it's `path` and `query`.

I think `return unless paging && next_page_link` is clearer than `return if !paging || next_page_path.nil?`.  If you prefer, I can change to something like this maybe?
```ruby
return unless paging
return unless next_page_link
```

Comment (bdunne) [reply lib/topological_inventory/orchestrator/worker.rb]: Also, I was thinking about dropping the whole conditional around paging, but didn't want to do that in this PR.  What do you think?  Both topo and sources should be paginated.

Comment (carbonin) [reply lib/topological_inventory/orchestrator/worker.rb]: > I didn't want to use path in the variable name, because it's not just a path, it's a relative URL. In our case it's path and query.

Makes sense. I'm okay with that logic.

> I think `return unless paging && next_page_link` is clearer than `return if !paging || next_page_path.nil?`

I'm okay with this too as long as the assignment makes its way out of the conditional.

> Also, I was thinking about dropping the whole conditional around paging, but didn't want to do that in this PR. What do you think? Both topo and sources should be paginated.

Both regular APIs are, but I added it for the internal APIs for tenants and authentications which I do not believe are paginated.
